packages/core/app-extensions/src/form/formModifier.js

```diff
@@ -5,6 +5,70 @@ export const ACTION_SAVE_ID = 'save'
 export const ACTION_GROUP_OUTPUT_ID = 'output'
 export const ACTION_GROUP_ACTIONS_ID = 'actions'
 
+export const removeBoxes = (formDefinition, boxIds) => ({
+  ...formDefinition,
+  children: formDefinition.children
+    .filter(item => item.componentType !== 'layout' || !boxIds.includes(item.id))
+    .map(item => {
+      if (item.componentType === 'layout') {
+        return removeBoxes(item, boxIds)
+      }
+
+      return item
+    })
+})
+
+export const removeActions = (formDefinition, actionIds) => ({
```

Comment:
you currently assume that an action is in an action group. i don't now if we want this assumption (possibly an action can be placed directly in the main action bar for widgets). Additionally if we have "action group A -> action group A1 -> action" we cannot remove the action.